controllers/api_v1/keyword.go

```diff
@@ -6,21 +6,52 @@ import (
 
 	"github.com/gutakk/go-google-scraper/helpers/api_helper"
 	helpers "github.com/gutakk/go-google-scraper/helpers/user"
+	"github.com/gutakk/go-google-scraper/services/keyword_api_service"
 	"github.com/gutakk/go-google-scraper/services/keyword_service"
 
 	"github.com/gin-gonic/gin"
 )
 
 const (
-	invalidFileErr = "invalid file"
+	invalidFileErr      = "invalid file"
+	keywordsNotFoundErr = "keywords not found"
 )
 
 type KeywordAPIController struct{}
 
 func (kapi *KeywordAPIController) ApplyRoutes(engine *gin.RouterGroup) {
+	engine.GET("/keywords", kapi.fetchKeywords)
 	engine.POST("/keywords", kapi.uploadKeyword)
 }
 
+func (kapi *KeywordAPIController) fetchKeywords(c *gin.Context) {
+	currentUserID := helpers.GetCurrentUserID(c)
+	keywordService := keyword_service.KeywordService{CurrentUserID: currentUserID}
+	keywords, err := keywordService.GetAll()
+
+	if err != nil {
+		errorResponse := api_helper.ErrorResponseObject{
+			Detail: err.Error(),
+			Status: http.StatusBadRequest,
+		}
+		c.JSON(errorResponse.Status, errorResponse.ConstructErrorResponse())
+		return
+	}
+
+	if len(keywords) == 0 {
+		errorResponse := api_helper.ErrorResponseObject{
+			Detail: errors.New(keywordsNotFoundErr).Error(),
+			Status: http.StatusNotFound,
```

Comment:
it's not really correct for the StatusNotFound error, it's a list, so if the list is empty then we will display an empty list, not  Found will be used in some `finding` case like `getting the detail keyword endpoint`.

Or `Not found` mean the user `finding something`, here is the list screen which is the user `query for the list` not `select for a object` so it's not NOT FOUND, what do you think @gutakk

Reply:
Agree with you. Here is the information from JSON API spec

> an array of resource objects, an array of resource identifier objects, or an empty array ([]), for requests that target resource collections

Reply:
Fixed in 647465c2157f2b4ba10af461ea9d51e0185e9090